### ISSUE TITLE
Rule Proposal: `no-v-show`

### ISSUE THREAD

**Issue by fisker** [vuejs/eslint-plugin-vue#1518]
**Please describe what the rule should do:**

I don't see a good reason not using html native `[hidden]` instead of `v-show` directive.
Even `hidden` used as a prop of component, it should behave like native one.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
<!-- BAD -->
<template>
	<div v-show="!showHide"></div>
</template>
```

```vue
<!-- GOOD -->
<template>
	<div :hidden="showHide"></div>
</template>
```

**Additional context**

N/A

**Comment by posva** [comment]
See https://github.com/vuejs/vue-next/issues/3973#issuecomment-863352310

This rule would complicate things more than help users since you still need v-show or v-if for transitions and having two version of doing the same thing depending on the parent component would be very confusing, specially when the transition component is outside of the vue file where the `v-show` is like when it's used at the root element of an SFC

**Comment by fisker** [comment]
> since you still need v-show or v-if for transitions

I don't think I ever need `v-show`, if it can't be enabled in the `recommend` preset, I'd like to enable myself.

Please reopen.

**Comment by ota-meshi** [comment]
You are free to ban any directives using the `vue/no-restricted-syntax` rule.

```json
{
  "vue/no-restricted-syntax": ["error", "VAttribute[directive=true][key.name.name=show]"]
}
```

https://eslint.vuejs.org/rules/no-restricted-syntax.html

**Comment by fisker** [comment]
Thanks, I need auto fix for it, I'll make my own plugin then.
